### PR TITLE
Feature/bull queue

### PR DESCRIPTION
**Pull request by mcousillas6**
#### Description:

* Added setup for background jobs using Bull
* Updated docker-compose to have redis setup.

#### Notes
* The PR targets the email service feature since it builds on top of that functionality

### CODE REVIEW

File: .env.sample [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
 DB_PORT=5432
 DB_USER=
 DB_PASSWORD=
 DB_NAME=node_base_dev
+
 SMTP_SERVICE=
 SMTP_USER=
 SMTP_PASS=
+
+REDIS_PORT=6379
```
Comment (mariosouto) [inline .env.sample:14]: Isn't it simpler if we just have a `REDIS_URL` var?

File: src/jobs/connection.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { QueueOptions } from 'bull';
```
Comment (mariosouto) [inline src/jobs/connection.ts]: Isn't this just the queue config more than the connection? Just talking about the naming of the file.

I think that you open the "real" connection when the queue is created

**Review by mariosouto — APPROVED** [review]
Good work, i left some smalls comments

File: .env.sample (hunk 1 above)
Comment (mcousillas6) [reply .env.sample:14]: I prefer having all three options separate, it think it's simpler to understand 🤷‍♂

Comment (mariosouto) [reply .env.sample:14]: Yeah, agree. I meant for handling deploys to prod. Generally when you set redis you get the full redis url, either in AWS or redis lab, so in those cases is just simpler to set `REDIS_URL` as an env var

**Review by santiestra — APPROVED** [review]

**Pull request merged**